src/callbacks_step/averaging_dg.jl

```diff
@@ -0,0 +1,58 @@
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+
+
+function save_averaging_file(averaging_callback, mesh::TreeMesh, equations, dg::DGSEM, cache)
+  @unpack output_directory, filename, mean_values = averaging_callback
+  h5open(joinpath(output_directory, filename), "w") do file
+    # Add context information
+    attributes(file)["ndims"] = ndims(mesh)
+    attributes(file)["polydeg"] = polydeg(dg)
+    attributes(file)["nelements"] = nelements(dg, cache)
+
+    # Store data as 1D arrays
+    file["v_mean"] = mean_values.v_mean[:]
+    file["c_mean"] = mean_values.c_mean[:]
+    file["rho_mean"] = mean_values.rho_mean[:]
+    file["vorticity_mean"] = mean_values.vorticity_mean[:]
+  end
+
+  return filename
+end
+
+
+function load_averaging_file(averaging_file, mesh::TreeMesh, equations, dg::DGSEM, cache)
+  # Read and check mesh and solver info
+  n_dims, n_nodes, n_elements = h5open(averaging_file, "r") do file
+    n_dims = read(attributes(file)["ndims"])
+    n_nodes = read(attributes(file)["polydeg"]) + 1
+    n_elements = read(attributes(file)["nelements"])
+
+    @assert n_dims == ndims(mesh) "ndims differs from value in averaging file"
+    @assert n_nodes - 1 == polydeg(dg) "polynomial degree in solver differs from value in averaging file"
+    @assert n_elements == nelements(dg, cache) "nelements in solver differs from value in averaging file"
+
+    return n_dims, n_nodes, n_elements
+  end
+
+  # Read data as 1D vectors
+  v_mean_vec, c_mean_vec, rho_mean_vec, vorticity_mean_vec = h5open(averaging_file, "r") do file
+    return read(file["v_mean"]),
+           read(file["c_mean"]),
+           read(file["rho_mean"]),
+           read(file["vorticity_mean"])
+  end
+
+  # Reshape arrays into solver-specific format
+  v_mean         = reshape(v_mean_vec,         (n_dims, repeat([n_nodes], n_dims)..., n_elements))
+  c_mean         = reshape(c_mean_vec,         (repeat([n_nodes], n_dims)..., n_elements))
+  rho_mean       = reshape(rho_mean_vec,       (repeat([n_nodes], n_dims)..., n_elements))
+  vorticity_mean = reshape(vorticity_mean_vec, (repeat([n_nodes], n_dims)..., n_elements))
```

Comment:
Sorry, I should have been clearer: I didn't want to store the arrays as multi-dimensional arrays, I was just wondering if you could *load* them in pre-allocated multi-dimensional arrays.

One advantage I see for storing them as flat arrays is that it would allow you to also convert them using `trixi2vtk`, at least if you also add the same information that `trixi2vtk` expects to find. But I think this is already much clearer, so let's leave it as it is.